### ISSUE TITLE
Istio with MTLS strict reporting certificate expired errors

### ISSUE THREAD

**Issue by hleal18** [istio/istio#38280]
### Bug Description

Using istio 1.12.4 with MTLS set to strict on GKE cluster v1.21.9-gke.300, we’re currently facing an issue with tcp connections failing to get established to destination pods after they’ve been running for quite some time (span of days and weeks). By enabling debug logs through `"sidecar.istio.io/logLevel": debug` we were able to spot the following TLS error: 

```
debug	envoy connection	[C8] TLS error: 268436501:SSL routines:OPENSSL_internal:SSLV3_ALERT_CERTIFICATE_EXPIRED
```

Inspecting the config dump reported by both client and destination pods we saw the entry for the certificate chain and by decoding it we saw that the certificate is still valid. So we don’t know exactly where this error is coming from.

Restarting the pod or setting mtls to disable helps to mitigate this issue and connections start working again, however, this is far from ideal.

Other issues that seem to be related: 

[https://github.com/istio/istio/issues/38081](https://github.com/istio/istio/issues/38081)

[https://github.com/istio/istio/issues/14942](https://github.com/istio/istio/issues/14942)

### Version

```prose
Istio version: 1.12.4
K8s version: v1.21.9-gke.300
```

**Comment by howardjohn** [comment]
Once it starts breaking does it break 100% of requests, or only some? Sounds like it never recovers without intervention, right? Any chance you waited >12hrs without intervention before and witnessed if it recovered?

Any chance you could look at the time as reported by the cluster nodes to verify it is accurate?

**Comment by hzxuzhonghu** [comment]
Can you paste the logs of sidedcar, you can strip the accesslog. There are some logs about cert create/rotate

**Comment by hleal18** [comment]
To give more context about our environment, this is affecting various databases that have been deployed for a few weeks already (right now 23 days+, but it could have started earlier without us noticing).

We’ve done more thorough tests about the amount of requests that fail and also trying other mechanisms to get around this issue with the databases that we use, in this case `postgres`, `mysql` and `mongodb`.

To answer specifically your question, once this issue starts, for `postgres` and `mongodb` the initial connection attempt from the client pod to the database will fail. We rewrote our pod command in a way that when the container starts, it doesn’t try to connect right away but sleeps for 60s instead and then attempts to connect. This worked well and connection was successfully established.  

Another experiment we did was to wrap the connection command from the client pod in a for loop with 5 iterations. In this case the first iteration always failed, but the remaining ones succeeded without issues.

For postgres failed connections the error reported is:

```jsx
psql: server closed the connection unexpectedly
This probably means the server terminated abnormally
```

And for mongodb:

```jsx
Failed: can't create session: could not connect to server: server selection error: server selection timeout, current topology: { Type: ReplicaSetNoPrimary, Servers: [{ Addr: mongo-0.mongodb-headless.namespace.svc.cluster.local:27017, Type: Unknown, Last error: connection() error occured during connection handshake: read tcp 10.4.4.118:47328->10.4.17.113:27017: read: connection reset by peer }, ] }
```

However, we also experimented the same way with various mysql databases that were showing this same symptoms (also 23+ days old). Adding the sleep or the for loop didn’t help to mitigate this issue. We also spawned a `mysql-client` pod, exec into it and tried to connect multiple times to the database. All of the connection attempts failed with the following error: `ERROR 2013 (HY000): Lost connection to MySQL server at 'reading initial communication packet', system error: 0` .

We then verified the istio debug logs, and all of databases reported the same error for failed connections `TLS error: 268436501:SSL routines:OPENSSL_internal:SSLV3_ALERT_CERTIFICATE_EXPIRED`. We also spotted the access logs for connection error with `UF, URX` codes: `[2022-04-08T18:28:14.492Z] "- - -" 0 UF,URX - - "-" 0 0 29 - "-" "-" "-" "-" "10.4.9.133:3306" outbound|3306||postgresql-primary.namespace.svc.cluster.local - 10.0.37.225:3306 10.4.4.38:54712 - -`

For all of the databases, either restarting it or disabling mtls helped to mitigate the situation. After that, client pods were able to connect to the database without the need of a sleep/retry mechanism.

The issue doesn’t go away on its own, looks like when it starts happening it’s there until either we disable mtls or restart the database.

Here I attach the logs for one of the postgres databases.
[Explore-logs-2022-04-08 14_08_36 (1).txt](https://github.com/istio/istio/files/8454876/Explore-logs-2022-04-08.14_08_36.1.txt)
